internal/datastore/test/tuples.go

```diff
@@ -89,64 +90,75 @@ func SimpleTest(t *testing.T, tester DatastoreTester) {
 				// Check that we can find the tuple a number of ways
 				rq := ds.ReverseQueryTuplesFromSubject(tupleToFind.User.GetUserset(), lastRevision)
 
-				queries := []datastore.CommonTupleQuery{
-					ds.QueryTuples(datastore.TupleQueryResourceFilter{
-						ResourceType:       tupleToFind.ObjectAndRelation.Namespace,
-						OptionalResourceID: tupleToFind.ObjectAndRelation.ObjectId,
-					}, lastRevision),
-					ds.QueryTuples(datastore.TupleQueryResourceFilter{
-						ResourceType: tupleToFind.ObjectAndRelation.Namespace,
-					}, lastRevision).WithUsersets([]*v0.ObjectAndRelation{tupleToFind.User.GetUserset()}),
-					ds.QueryTuples(datastore.TupleQueryResourceFilter{
-						ResourceType:             tupleToFind.ObjectAndRelation.Namespace,
-						OptionalResourceID:       tupleToFind.ObjectAndRelation.ObjectId,
-						OptionalResourceRelation: tupleToFind.ObjectAndRelation.Relation,
-					}, lastRevision),
-					ds.QueryTuples(datastore.TupleQueryResourceFilter{
-						ResourceType:       tupleToFind.ObjectAndRelation.Namespace,
-						OptionalResourceID: tupleToFind.ObjectAndRelation.ObjectId,
-					}, lastRevision).WithUsersets([]*v0.ObjectAndRelation{tupleToFind.User.GetUserset()}),
-					ds.QueryTuples(datastore.TupleQueryResourceFilter{
-						ResourceType:             tupleToFind.ObjectAndRelation.Namespace,
-						OptionalResourceRelation: tupleToFind.ObjectAndRelation.Relation,
-					}, lastRevision).WithUsersets([]*v0.ObjectAndRelation{tupleToFind.User.GetUserset()}),
-					ds.QueryTuples(datastore.TupleQueryResourceFilter{
-						ResourceType:             tupleToFind.ObjectAndRelation.Namespace,
-						OptionalResourceRelation: tupleToFind.ObjectAndRelation.Relation,
-					}, lastRevision).WithUsersets([]*v0.ObjectAndRelation{tupleToFind.User.GetUserset()}).Limit(1),
-
-					rq.WithObjectRelation(tupleToFind.ObjectAndRelation.Namespace, tupleToFind.ObjectAndRelation.Relation),
-					rq.WithObjectRelation(tupleToFind.ObjectAndRelation.Namespace, tupleToFind.ObjectAndRelation.Relation).Limit(1),
-				}
-				for _, query := range queries {
-					iter, err := query.Execute(ctx)
-					require.NoError(err)
-					tRequire.VerifyIteratorResults(iter, tupleToFind)
-				}
+				iter, err := ds.QueryTuples(ctx, &v1.RelationshipFilter{
+					ResourceType:       tupleToFind.ObjectAndRelation.Namespace,
+					OptionalResourceId: tupleToFind.ObjectAndRelation.ObjectId,
+				}, lastRevision)
+				require.NoError(err)
+				tRequire.VerifyIteratorResults(iter, tupleToFind)
```

Comment:
Can we add some negative tests here, where the tuple is not found due to the extra options given?